pkg/api/store/password/store.go

```diff
@@ -0,0 +1,276 @@
+	buildFieldData(convert.ToMapInterface(p.Fields[id]), data, &fieldData, path)
+
+	if err := p.handlePasswordFields(fieldData, data, id); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *PasswordStore) assignBack(data map[string]interface{}, id string) error {
+	var fieldData []fieldInfo
+	var path []string
+
+	buildFieldData(convert.ToMapInterface(p.Fields[id]), data, &fieldData, path)
+
+	for _, info := range fieldData {
+		split := strings.SplitN(info.value, ":", 2)
+		if len(split) != 2 {
+			continue
+		}
+		value, err := p.getSecret([]string{"mgmt-secrets", "githubconfig-clientsecret"})
+		if err != nil {
+			return fmt.Errorf("error getting secret for field %s", info.value)
+		}
+		values.PutValue(data, value, info.paths...)
+	}
+	return nil
+}
+
+func (p *PasswordStore) handlePasswordFields(fieldData []fieldInfo, data map[string]interface{}, id string) error {
+	for _, info := range fieldData {
+		var name, namespace string
+		if _, ok := data["name"]; ok {
+			name = convert.ToString(data["name"])
+		} else {
+			name = fmt.Sprintf("%s-%s", id, info.paths[len(info.paths)-1])
+		}
+		if val, ok := data["id"]; ok {
+			splitID := strings.Split(convert.ToString(val), ":")
+			if len(splitID) == 2 {
+				namespace = splitID[1]
+			} else if len(splitID) == 1 {
+				namespace = splitID[0]
+			}
+		}
+		if namespace == "" {
+			namespace = "mgmt-secrets"
+		}
+		if err := p.createOrUpdateSecrets(info.value, name, namespace); err != nil {
+			return err
+		}
+		values.PutValue(data, fmt.Sprintf("%s:%s", namespace, strings.ToLower(name)), info.paths...)
+	}
+	return nil
+}
+
+func buildFieldData(data1 map[string]interface{}, data2 map[string]interface{}, fieldData *[]fieldInfo, path []string) {
+	for key1, val1 := range data1 {
+		if val2, ok := data2[key1]; ok {
+			if convert.ToString(val1) == ".." {
+				val := convert.ToString(val2)
+				if val != "" {
+					split := strings.Split(val, ":")
+					if len(split) == 2 && (split[0] == "mgmt-secrets") {
+						// rancher referenced
+						logrus.Debugf("rancher referenced, continue %s", val)
+						continue
+					}
+				}
+				path = append(path, key1)
+				*fieldData = append(*fieldData, fieldInfo{path, val})
+			} else {
+				valArr := convert.ToMapSlice(val2)
+				if valArr == nil {
+					buildFieldData(convert.ToMapInterface(val1), convert.ToMapInterface(val2), fieldData, append(path, key1))
+				} else {
+					for _, each := range valArr {
+						buildFieldData(convert.ToMapInterface(val1), each, fieldData, append(path, key1))
+					}
+				}
+			}
+		}
+	}
+}
+
+func (p *PasswordStore) createOrUpdateSecrets(data, name, namespace string) error {
+	_, err := p.nsStore.Get(namespace, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		if _, err = p.nsStore.Create(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}); err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	if err != nil {
+		return err
```

Comment:
looks like we would return here even if the error is not found, and we create a secret?

Reply:
if error is not found, I create the namespace.. after that returning all errors except already exists

Reply:
@kinarashah don't we need to create a secret?